### PR TITLE
Connect multi

### PR DESCRIPTION
**Pull request by JorisMonnet**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/blindwar/game/util/Util.kt [hunk 1]
```diff
@@ -69,10 +67,25 @@ object Util {
...
-    fun updateProfileImage(liveData: LiveData<StorageReference>,
-                           imageView: ImageView,
-                           viewLifecycleOwner: LifecycleOwner,
-                           context: Context) {
+    /**
+     * Check if internet is working via one ping
+     *
+     * @return
+     */
+    fun isOnline(): Boolean {
```
Comment (ghemDD) [inline app/src/main/java/ch/epfl/sdp/blindwar/game/util/Util.kt:75]: Useful function ! We should make use of it to create a NetworkConnectivtyChecker that would signal the current connectivity state of an user.

File: app/src/main/java/ch/epfl/sdp/blindwar/game/util/DisplayableItemAdapter.kt [hunk 2]
```diff
@@ -215,6 +241,50 @@ class DisplayableItemAdapter(
...
+            builder.setPositiveButton(
+                context.getString(R.string.cancel_btn)
+            ) { it, _ -> it.cancel() }
+            builder.setOnCancelListener {
+                Toast.makeText(
+                    context,
+                    context.getString(R.string.toast_canceled_match_creation),
+                    Toast.LENGTH_SHORT
+                ).show()
+                MatchDatabase.removeMatch(matchUID, Firebase.firestore)
```
Comment (ghemDD) [inline app/src/main/java/ch/epfl/sdp/blindwar/game/util/DisplayableItemAdapter.kt:261]: May break MVVM pattern

File: app/src/main/java/ch/epfl/sdp/blindwar/game/util/DisplayableItemAdapter.kt [hunk 3]
```diff
@@ -316,6 +386,44 @@ class DisplayableItemAdapter(
         const val DURATION_FAST = 20000L
         const val DURATION_DEFAULT = 30000L
     }
+
```
Comment (ghemDD) [inline app/src/main/java/ch/epfl/sdp/blindwar/game/util/DisplayableItemAdapter.kt:389]: Try to move the share logic in a separate file to reduce the number of lines in this file

**Review by ghemDD — APPROVED** [review]
Very important additions thanks 
Nice formatting

File: app/src/androidTest/java/ch/epfl/sdp/blindwar/database/UserDatabaseTest.kt [hunk 4]
```diff
@@ -139,7 +140,8 @@ class UserDatabaseTest : TestCase() {
     @Test
     fun getUserStatisticsTest() {
         launchFragmentInContainer<ProfileFragment>()
-        val fail = 1; val score = 2
+        val fail = 1;
```
Comment (Marty99-dev) [inline app/src/androidTest/java/ch/epfl/sdp/blindwar/database/UserDatabaseTest.kt:143]: Trailing ";"

**Review by Marty99-dev — APPROVED** [review]

**Pull request merged**